datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/TestKafkaConnectorTask.java

```diff
@@ -170,7 +171,58 @@ public void testConsumeWithStartingOffset() throws Exception {
     DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
     task.setEventProducer(datastreamProducer);
 
-    KafkaConnectorTask connectorTask = createKafkaConnectorTask(task);
+    KafkaConnectorTask connectorTask = createKafkaConnectorTaskWithAutoOffsetResetConfig(task, "earliest");
+
+    // validate auto.offset.reset config is overridden to none (given the start offsets)
+    Assert.assertEquals(connectorTask.getConsumerAutoOffsetResetConfig(), "none");
+
+
+    LOG.info("Sending third set of events");
+
+    //send 100 more msgs
+    produceEvents(_kafkaCluster, _zkUtils, topic, 1000, 100);
+
+    if (!PollUtils.poll(() -> datastreamProducer.getEvents().size() == 200, 100, POLL_TIMEOUT_MS)) {
+      Assert.fail("did not transfer 200 msgs within timeout. transferred " + datastreamProducer.getEvents().size());
+    }
+
+    connectorTask.stop();
+    Assert.assertTrue(connectorTask.awaitStop(CONNECTOR_AWAIT_STOP_TIMEOUT_MS, TimeUnit.MILLISECONDS),
+        "did not shut down on time");
+  }
+
+  @Test
+  public void testConsumeWithStartingOffsetAndResetStrategy() throws Exception {
+    String topic = "pizza1";
+    createTopic(_zkUtils, topic);
+
+    LOG.info("Sending first set of events");
+
+    //produce 100 msgs to topic before start
+    produceEvents(_kafkaCluster, _zkUtils, topic, 0, 100);
+    Map<Integer, Long> startOffsets = Collections.singletonMap(0, 100L);
+
+    LOG.info("Sending second set of events");
+
+    //produce 100 msgs to topic before start
+    produceEvents(_kafkaCluster, _zkUtils, topic, 100, 100);
+
+    //start
+    MockDatastreamEventProducer datastreamProducer = new MockDatastreamEventProducer();
+    Datastream datastream = getDatastream(_broker, topic);
+
+    // Unable to set the start position, OffsetToTimestamp is returning null in the embedded Kafka cluster.
```

Comment:
The name start position unfortunately is confusing since it is a timestamp. Since you have set it to 100 I am guessing it is returning null? And so what happens when it returns null? What is the code flow when this happens?

Reply:
a) I copied and modfied `testConsumeWithStartOffset`, so not sure what this comment refers to, I removed it.
b) it is not a timestamp, it is an offset. The metadata key is: `DatastreamMetadataConstants.START_POSITION = "system.start.position"`

Reply:
I see. So it does seem to be offset in the implementation and there are some other connectors storing time. I was guessing that is where the comment about offsetForTimestamp was coming from. Good to know it actually causes no issues.